numactl.c

```diff
@@ -507,11 +513,18 @@ int main(int ac, char **av)
 
 			errno = 0;
 			did_node_cpu_parse = 1;
-			setpolicy(MPOL_INTERLEAVE);
+			if (weighted_interleave)
+				setpolicy(MPOL_WEIGHTED_INTERLEAVE);
+			else
+				setpolicy(MPOL_INTERLEAVE);
```

Comment:
I worry this requires `--weighted`/`-w` option must be used before `--interleave`/`-i` option. I'm not sure if it's possible to make the order independant with the current option system.

Reply:
I looked at this for a little bit and decided to just get something up rather than fuss over it too much.

Basically choose 1
1. Basically duplicate the exact same logic from --interleave in the switch
2. add a -w switch that adds to the existing --interleave logic

The latter does have an ordering issue, and the code is not well designed to manage that very well.  If the first option is preferably, this is like 10 minutes of work, just need to know what the maintainers prefer.

Reply:
function is simple enough, i didn't want to rip apart the other code just to hack this in, but i'll take a look at it tomorrow if i can.

Reply:
actually, even dumber idea:  just have an explicit fall-through in the `-w`

Reply:
> actually, even dumber idea: just have an explicit fall-through in the -w option. Done.

Make sense. It'd be better to have a logic that warns users when both options are used, but it's also fine without it.